### PR TITLE
Security: DepthwiseConv2dNativeBackpropFilter returns uninitialized heap memory when out_backprop is empty (CWE-908 information disclosure)

### PR DESCRIPTION
**Pull request by endorphin13**
`DepthwiseConv2dNativeBackpropFilterOp::Compute` (`tensorflow/core/kernels/depthwise_conv_grad_op.cc:1145-1151`) sizes its output from `filter_shape` but takes its early return on a **different** tensor:

```cpp
OP_REQUIRES_OK(context, context->forward_input_or_allocate_output(
                            {1}, 0, filter_shape, &filter_backprop));

// If there is nothing to compute, return.
if (out_backprop.shape().num_elements() == 0) {
  return;
}
```

`filter_shape` and `out_backprop` are independent. An `out_backprop` with a zero-length dimension therefore leaves a **non-empty `filter_backprop` that nothing has written**. `forward_input_or_allocate_output` does not initialize the buffer, and the forwarding candidate is input 1 (`filter_sizes`, `int32`) for a `float` output — so no input is ever forwarded and the allocation is always fresh. The caller receives whatever the allocator last held.

Affects `float`, `double`, `half` and `bfloat16` on CPU. Present on current master.

## Why this is a security issue

**CWE-908** (use of uninitialized resource) leading to **CWE-200** (information exposure). It is **not** memory corruption — the read stays inside a correctly-sized allocation, there is no out-of-bounds access and no write. That is also why sanitizers have not caught it: TF's OSS-Fuzz configuration builds with the address and undefined sanitizers, neither of which tracks definedness of bytes.

## Demonstrated impact

I have working end-to-end proof-of-concept code for everything below. **It is deliberately not included here**, and I will provide it privately through the Google Bug Hunters process.

**Reachable through ordinary Keras models, not a synthetic graph.** The trigger is an empty batch or a zero-length spatial dimension — ordinary runtime input values, no graph edit, no attribute change. It reproduces through unmodified `tf.keras.applications` architectures: **Xception** (34 of 34 separable kernels) and **NASNetMobile** (159 of 160), and through plain `tf.keras.layers.DepthwiseConv2D` / `SeparableConv2D` / `DepthwiseConv1D` / `SeparableConv1D`.

**The disclosure is persisted into model weights and is recoverable.** The leaked buffer is a weight gradient, so an optimizer writes it into the model:

- with an empty batch, the correct gradient is exactly zero, yet the model's weights change on **12 of 12** trials, with deltas up to 1e33 — silent model corruption with no error raised;
- because `w_after = w_before - lr * g`, the leaked values are recovered exactly by `(w_before - w_after) / lr` on **10 of 12** trials at realistic learning rates (0.01, 0.001). Anyone who receives the fine-tuned weights — which is the product of a hosted fine-tuning service — can perform that subtraction.

**It crosses process and privilege boundaries.** Where workers are created with `fork()` (gunicorn `--preload`, uWSGI, `multiprocessing`'s default start method on Linux), a child process recovers data the parent handled and freed, at **20/20**. Where that worker drops privileges before handling untrusted input, an **unprivileged process reads memory only the privileged parent ever held**, also at **20/20** — `setuid` does not scrub the inherited address space. Recovery is byte-exact: verified by SHA-256 against planted contents including weight matrices, credential-shaped blobs and PII-shaped records. Every differential control is clean — the marker never appears when no prior request ran.

Bounded honestly: it does **not** reach unrelated processes, containers or VMs (the kernel zeroes pages handed to a fresh address space), and it cannot lead to code execution — there is no write primitive, no attacker-controlled index and no influence on control flow. It is an information-disclosure and integrity primitive.

## The fix

The 3-D sibling already handles this exact situation, at `conv_grad_filter_ops_3d.cc:236` and `:376`:

```cpp
filter_backprop->template flat<T>().setZero();
```

This applies the same treatment. A filter that received no contribution has a gradient of exactly zero.

**Behaviour is unchanged for every defined input:**

| case | before | after |
|---|---|---|
| `filter_shape` empty | returns empty output | identical — the new guard is false |
| `out_backprop` non-empty | computes normally | identical — the branch is not taken |
| `out_backprop` empty, `filter_shape` non-empty | returns **uninitialized** memory | returns zeros |

Only previously-undefined values change, so no correct program can regress, and the non-degenerate path is untouched.

The `DepthwiseConv2DBackpropInput` early return at `:637` is **not** affected — it allocates from `input_shape` and guards on the same tensor, so its output is genuinely empty when the guard fires.

## Tests

No test added here — I would value guidance on placement. `DepthwiseConv2dNativeBackpropFilter` with an `out_backprop` containing a zero-length dimension and a non-empty `filter_sizes` should return an all-zero tensor; today it returns whatever the allocator last held. Glad to add that in this PR.

Not compile-tested (no Bazel environment) — please run CI.

### CODE REVIEW

**Review by dmiltr3 — APPROVED** [review]

**Comment by endorphin13** [comment]
Hello, do you need any additional changes? Wanted to learn when this is going to be merged.
